addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -247,14 +247,46 @@ class EnvironmentMountService extends Service {
     const disAllowedUsers = this._getDisAllowedUsers(updateRequest);
     const permissionChangeUsers = this._getPermissionChangeUsers(updateRequest);
 
-    if (allowedUsers.length) {
-      await this.addPermissions(allowedUsers, studyId);
+    if (_.isEmpty(allowedUsers) && _.isEmpty(disAllowedUsers) && _.isEmpty(permissionChangeUsers)) {
+      return;
     }
-    if (disAllowedUsers.length) {
-      await this.removePermissions(disAllowedUsers, studyId);
+
+    const total = _.size(allowedUsers) + _.size(disAllowedUsers) + _.size(permissionChangeUsers);
+    if (total > 1000) {
+      this.boom.internalError(
+        `Cannot update ${total} workspace permissions at a time. Please reduce the number of user permission changes and try again`,
+      );
     }
-    if (permissionChangeUsers.length) {
-      await this.updatePermissions(permissionChangeUsers, studyId, updateRequest);
+
+    const errors = [];
+
+    if (!_.isEmpty(allowedUsers)) {
+      const errorsforAdd = await this.addPermissions(allowedUsers, studyId);
```

Comment:
You can do this differently.  You can make the this.addPermissions throws an exception that contains the errors.

Reply:
Ah, yes. That makes more sense.